UsbAccess/UsbAccess.cpp

```diff
@@ -769,5 +1064,71 @@ namespace Plugin {
 
         return result;
     }
+
+    uint32_t UsbAccess::getUSBSizeWrapper(const JsonObject &parameters, JsonObject &response)
+    {
+        LOGINFOMETHOD();
+
+        bool result = false;
+        char cmd[128] = {'\0'};
+        char totalsize[20] = {'\0'};
+        char usedsize[20] = {'\0'};
+        char freesize[20] = {'\0'};
+
+        string pathParam;
+        if (parameters.HasLabel("path"))
+        {
+            pathParam = parameters["path"].String();
+        }
+        else
+        {
+            response["error"] = "don't specify the mount path";
+            returnResponse(result);
+        }
+        snprintf(cmd, 127, "df -h | grep %s$ | awk \'{print $2 \" \" $3 \" \" $4}\'", pathParam.c_str());
+        FILE* fp = popen(cmd, "r");
```

Comment:
Since the popen is used on the data coming from outside, v_secure_popen() should be used for handling more secure way.

Reply:
Not able to get response when using **v_secure_popen()**. Can you please provide more inputs on the usage of this method?

```
root@HP44HRDK:~# curl -d '{"jsonrpc":"2.0","id":42,"method": "org.rdk.UsbAccess.getUSBSize","params":{"path":"\/run\/media\/sda1"}}' http://127.0.0.1:9998/jsonrpc
Jun 21 03:37:28 HP44HRDK WPEFramework[5315]: [5319] INFO [UsbAccess.cpp:1075] getUSBSizeWrapper: params={"path":"\/run\/media\/sda1"}
Jun 21 03:37:28 HP44HRDK WPEFramework[8167]: /run/media/sda1: Permission denied
{"jsonrpc":"2.0","id":42,"result":{"success":false}}root@HP44HRDK:~# Jun 21 03:37:28 HP44HRDK WPEFramework[5315]: [5319] INFO [UsbAccess.cpp:1099] getUSBSizeWrapper: response={"success":false}
root@HP44HRDK:~#
```

Reply:
Done.